src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionsState.cs

```diff
@@ -74,25 +76,87 @@ public SubscriptionConnectionsState(long subscriptionId, SubscriptionStorage sto
             _waitForMoreDocuments = new AsyncManualResetEvent(CancellationTokenSource.Token);
         }
 
-        public void Initialize(SubscriptionConnection connection, bool afterSubscribe = false)
+        public async Task InitializeAsync(SubscriptionConnection connection, bool afterSubscribe = false)
         {
             _subscriptionName = connection.Options.SubscriptionName ?? _subscriptionId.ToString();
-            _subscriptionState = connection.SubscriptionState;
             Query = connection.SubscriptionState.Query;
 
+            if (afterSubscribe == false)
+                return;
+
             // update the subscription data only on new concurrent connection or regular connection
-            if (afterSubscribe && _connections.Count == 1)
+            if (IsConcurrent == false)
+            {
+                RefreshFeatures(connection);
+                return;
+            }
+
+            DocumentDatabase.ForTestingPurposes?.Subscription_ActionToCallDuringWaitForSubscribe?.Invoke(_connections);
```

Comment:
this is only for concurrent subscription? let's add this to the name

Reply:
done